valohai_yaml/objs/pipelines/node.py

```diff
@@ -43,6 +43,8 @@ def parse_qualifying(cls, data: dict) -> 'Node':
     def serialize(self) -> dict:
         ser = super().serialize()
         ser['type'] = self.type
+        if not ser.get('actions'):
+            ser.pop('actions', None)
```

Comment:
if `actions` falsy or non-existent, **try** to remove `actions` from the dict

the code could signify intent a bit more but that is mainly a personal preference :thinking: 

such a tiny snippet anyway + a test to back it up